src/tasks/index.ts

```
@@ -0,0 +1,174 @@
+			if (
+				request.values.is_scheduled_with__cron_expression != null &&
+				request.values.is_scheduled_to_execute_on__time == null
+			) {
+				try {
+					request.values.is_scheduled_to_execute_on__time = cronParser
+						.parseExpression(request.values.is_scheduled_with__cron_expression)
+						.next()
+						.toDate()
+						.toISOString();
+				} catch (_) {
+					throw new BadRequestError(
+						`Invalid cron expression: ${request.values.is_scheduled_with__cron_expression}`,
+					);
+				}
+			}
+
+			// Assert that the provided start time is far enough in the future
+			if (request.values.is_scheduled_to_execute_on__time != null) {
+				const now = new Date(new Date().getTime() + tasksEnv.queueIntervalMS);
+				const startTime = new Date(
+					request.values.is_scheduled_to_execute_on__time,
+				);
+				if (startTime < now) {
+					throw new BadRequestError(
+						`Task scheduled start time must be greater than ${tasksEnv.queueIntervalMS} milliseconds in the future`,
+					);
+				}
+			}
+
+			// Assert that the requested handler exists
+			const handlerName = request.values.is_executed_by__handler;
+			if (handlerName == null) {
+				throw new BadRequestError(`Must specify a task handler to execute`);
+			}
+			const handler = worker?.handlers[handlerName];
+			if (handler == null) {
+				throw new BadRequestError(
+					`No task handler with name '${handlerName}' registered`,
+				);
+			}
+
+			// Assert that the provided parameter set is valid
+			if (handler.validate != null) {
+				if (!handler.validate(request.values.is_executed_with__parameter_set)) {
+					throw new BadRequestError(
+						`Invalid parameter set: ${ajv.errorsText(handler.validate.errors)}`,
+					);
+				}
+			}
+		},
+	});
+	worker.start();
+}
+
+// Register a task handler
+export function addTaskHandler(
+	name: string,
+	fn: TaskHandler['fn'],
+	schema?: Schema,
+): void {
+	if (worker == null) {
+		return;
+	}
```

I assume this is because we only support tasks on postgres, which is fine but should throw an error here rather than silently/surprisingly doing nothing